pkg/operator/apiservercontrollerset/apiservercontrollerset.go

```diff
@@ -0,0 +1,179 @@
+package apiservercontrollerset
+
+import (
+	"context"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned"
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/apiservicecontroller"
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/nsfinalizercontroller"
+	prune "github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/prunecontroller"
+	"github.com/openshift/library-go/pkg/operator/encryption"
+	"github.com/openshift/library-go/pkg/operator/encryption/controllers/migrators"
+	encryptiondeployer "github.com/openshift/library-go/pkg/operator/encryption/deployer"
+	"github.com/openshift/library-go/pkg/operator/events"
+	libgoresourcesyncer "github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/revisioncontroller"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/revision"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	apiregistrationclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	apiregistrationinformers "k8s.io/kube-aggregator/pkg/client/informers/externalversions"
+)
+
+type APIServerControllerSet struct {
+	targetNamespace string
+
+	kubeClient              kubernetes.Interface
+	operatorClient          v1helpers.OperatorClient
+	apiserverOperatorClient v1helpers.OperatorClient
+
+	apiServiceController  *apiservicecontroller.APIServiceController
+	encryptionControllers *encryption.Controllers
+	finalizerController   *nsfinalizercontroller.FinalizerController
+	pruneController       *prune.PruneController
+	revisionController    *revisioncontroller.RevisionController
+
+	eventRecorder events.Recorder
+}
+
+func NewAPIServerControllerSet(
```

Comment:
I expected this to be nearly empty.

Reply:
I couldn't really decide whether to hide certain arguments that will otherwise repeat in the "With" calls, I'm fine with just having a controllers slice here + have attributes for controllers that don't include `Run(workers int, stopChan <-chan struct{})` in their interface.

Reply:
I sticked with having just the `operatorClient` and `eventsRecorder` appear in this struct, I expect that at least the former should be the same for all the controllers appearing here, and there is probably no reason to have different event recorders either.

Reply:
acceptable